### DISCUSSION TITLE
Modify

### DISCUSSION THREAD

**Discussion by c0sm3s** [cantools/cantools#315] [Q&A] [answered]
Hi, may I get you help to modify the code to do the following.
I want the `decode_message()` method to give the decoded CAN message in either decimal or hexadecimal instead of the symbolic values of the signal.
For example if I do the following:
`decoded_values = (db.decode_message(946, b'\x44\x58\x00\x00\x00\x00\x00\x00'))`

I get something like this:
`{'Ignition_Status': 'Run', 'Parklamp_Status': 'On', Litval': 'Night', ...}`

But I want something like this:
`{'Ignition_Status': 0x1, 'Parklamp_Status': 0x1, Litval': 0xA, ...}`
Or
`{'Ignition_Status': 1, 'Parklamp_Status': 1, Litval': 10, ...}`

I've tried to look for the code section that does this but I couldn't find it, I'm assuming I need to skip the section that uses the VAL_TABLE_ from the DBC file and return that value. 
Thanks.

**Comment by juleq** [comment, marked as answer]
That is already supported. Look for the doc on decode_message:

`decode_message(frame_id_or_name, data, decode_choices=True, scaling=True)`

"Decode given signal data data as a message of given frame id or name frame_id_or_name. Returns a dictionary of signal name-value entries.

If decode_choices is False scaled values are not converted to choice strings (if available)."

**Reply by c0sm3s** [reply]
Thanks!!
